### ISSUE TITLE
`ogc` fails during `mage integration:test` with a number quota and resouce errors

### ISSUE THREAD

**Issue by fearful-symmetry** [elastic/elastic-agent#2806]
Right now, running `mage integration:test` produces a very long python traceback.

The core error lines are:
```
libcloud.common.google.QuotaExceededError: "Quota 'FIREWALLS' exceeded.  Limit: 500.0 globally."
```

```
libcloud.common.google.ResourceNotFoundError: {'message': "The resource 'projects/elastic-platform-ingest/global/firewalls/ci0co38vaiic8uihhreg' was not found", 'domain': 'global', 'reason': 'notFound'}
```

The full traceback is:

```
Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 358, in create_firewall
    self.provisioner.ex_get_firewall(name)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 7628, in ex_get_firewall
    response = self.connection.request(request, method="GET").object
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 285, in parse_body
    raise ResourceNotFoundError(message, self.status, code)
libcloud.common.google.ResourceNotFoundError: {'message': "The resource 'projects/elastic-platform-ingest/global/firewalls/ci0co38vaiic8uihhre0' was not found", 'domain': 'global', 'reason': 'notFound'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "/venv/lib/python3.11/site-packages/ogc/deployer.py", line 214, in _up_async
    provisioner.setup()
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 315, in setup
    self.create_firewall(self.layout.name, self.layout.ports, tags)
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 361, in create_firewall
    self.provisioner.ex_create_firewall(  # type: ignore
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 3577, in ex_create_firewall
    self.connection.async_request(request, method="POST", data=firewall_data)
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 871, in async_request
    response = request(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 267, in parse_body
    raise QuotaExceededError(message, self.status, code)
libcloud.common.google.QuotaExceededError: "Quota 'FIREWALLS' exceeded.  Limit: 500.0 globally."
2023-06-07T18:27:43Z <Greenlet at 0x7f056a3a2d40: _up_async(<LayoutModel: 5>)> failed with QuotaExceededError

Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 358, in create_firewall
    self.provisioner.ex_get_firewall(name)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 7628, in ex_get_firewall
    response = self.connection.request(request, method="GET").object
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 285, in parse_body
    raise ResourceNotFoundError(message, self.status, code)
libcloud.common.google.ResourceNotFoundError: {'message': "The resource 'projects/elastic-platform-ingest/global/firewalls/ci0co38vaiic8uihhrbg' was not found", 'domain': 'global', 'reason': 'notFound'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "/venv/lib/python3.11/site-packages/ogc/deployer.py", line 214, in _up_async
    provisioner.setup()
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 315, in setup
    self.create_firewall(self.layout.name, self.layout.ports, tags)
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 361, in create_firewall
    self.provisioner.ex_create_firewall(  # type: ignore
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 3577, in ex_create_firewall
    self.connection.async_request(request, method="POST", data=firewall_data)
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 871, in async_request
    response = request(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 267, in parse_body
    raise QuotaExceededError(message, self.status, code)
libcloud.common.google.QuotaExceededError: "Quota 'FIREWALLS' exceeded.  Limit: 500.0 globally."
2023-06-07T18:27:44Z <Greenlet at 0x7f056a3a2a20: _up_async(<LayoutModel: 3>)> failed with QuotaExceededError

Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 358, in create_firewall
    self.provisioner.ex_get_firewall(name)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 7628, in ex_get_firewall
    response = self.connection.request(request, method="GET").object
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 285, in parse_body
    raise ResourceNotFoundError(message, self.status, code)
libcloud.common.google.ResourceNotFoundError: {'message': "The resource 'projects/elastic-platform-ingest/global/firewalls/ci0co38vaiic8uihhrc0' was not found", 'domain': 'global', 'reason': 'notFound'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "/venv/lib/python3.11/site-packages/ogc/deployer.py", line 214, in _up_async
    provisioner.setup()
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 315, in setup
    self.create_firewall(self.layout.name, self.layout.ports, tags)
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 361, in create_firewall
    self.provisioner.ex_create_firewall(  # type: ignore
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 3577, in ex_create_firewall
    self.connection.async_request(request, method="POST", data=firewall_data)
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 871, in async_request
    response = request(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 267, in parse_body
    raise QuotaExceededError(message, self.status, code)
libcloud.common.google.QuotaExceededError: "Quota 'FIREWALLS' exceeded.  Limit: 500.0 globally."
2023-06-07T18:27:44Z <Greenlet at 0x7f056a3a2ac0: _up_async(<LayoutModel: 4>)> failed with QuotaExceededError

Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 358, in create_firewall
    self.provisioner.ex_get_firewall(name)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 7628, in ex_get_firewall
    response = self.connection.request(request, method="GET").object
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 285, in parse_body
    raise ResourceNotFoundError(message, self.status, code)
libcloud.common.google.ResourceNotFoundError: {'message': "The resource 'projects/elastic-platform-ingest/global/firewalls/ci0co38vaiic8uihhr90' was not found", 'domain': 'global', 'reason': 'notFound'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "/venv/lib/python3.11/site-packages/ogc/deployer.py", line 214, in _up_async
    provisioner.setup()
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 315, in setup
    self.create_firewall(self.layout.name, self.layout.ports, tags)
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 361, in create_firewall
    self.provisioner.ex_create_firewall(  # type: ignore
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 3577, in ex_create_firewall
    self.connection.async_request(request, method="POST", data=firewall_data)
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 871, in async_request
    response = request(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 267, in parse_body
    raise QuotaExceededError(message, self.status, code)
libcloud.common.google.QuotaExceededError: "Quota 'FIREWALLS' exceeded.  Limit: 500.0 globally."
2023-06-07T18:27:44Z <Greenlet at 0x7f056a3a2b60: _up_async(<LayoutModel: 1>)> failed with QuotaExceededError

Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 358, in create_firewall
    self.provisioner.ex_get_firewall(name)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 7628, in ex_get_firewall
    response = self.connection.request(request, method="GET").object
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 285, in parse_body
    raise ResourceNotFoundError(message, self.status, code)
libcloud.common.google.ResourceNotFoundError: {'message': "The resource 'projects/elastic-platform-ingest/global/firewalls/ci0co38vaiic8uihhreg' was not found", 'domain': 'global', 'reason': 'notFound'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "/venv/lib/python3.11/site-packages/ogc/deployer.py", line 214, in _up_async
    provisioner.setup()
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 315, in setup
    self.create_firewall(self.layout.name, self.layout.ports, tags)
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 361, in create_firewall
    self.provisioner.ex_create_firewall(  # type: ignore
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 3577, in ex_create_firewall
    self.connection.async_request(request, method="POST", data=firewall_data)
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 871, in async_request
    response = request(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 267, in parse_body
    raise QuotaExceededError(message, self.status, code)
libcloud.common.google.QuotaExceededError: "Quota 'FIREWALLS' exceeded.  Limit: 500.0 globally."
2023-06-07T18:27:44Z <Greenlet at 0x7f056a3a2ca0: _up_async(<LayoutModel: 6>)> failed with QuotaExceededError

Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 358, in create_firewall
    self.provisioner.ex_get_firewall(name)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 7628, in ex_get_firewall
    response = self.connection.request(request, method="GET").object
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 285, in parse_body
    raise ResourceNotFoundError(message, self.status, code)
libcloud.common.google.ResourceNotFoundError: {'message': "The resource 'projects/elastic-platform-ingest/global/firewalls/ci0co38vaiic8uihhr9g' was not found", 'domain': 'global', 'reason': 'notFound'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "/venv/lib/python3.11/site-packages/ogc/deployer.py", line 214, in _up_async
    provisioner.setup()
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 315, in setup
    self.create_firewall(self.layout.name, self.layout.ports, tags)
  File "/venv/lib/python3.11/site-packages/ogc/provision.py", line 361, in create_firewall
    self.provisioner.ex_create_firewall(  # type: ignore
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 3577, in ex_create_firewall
    self.connection.async_request(request, method="POST", data=firewall_data)
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 871, in async_request
    response = request(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/compute/drivers/gce.py", line 164, in request
    response = super().request(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 934, in request
    return super().request(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 646, in request
    return request_to_be_executed(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 708, in _retryable_request
    response = responseCls(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/base.py", line 160, in __init__
    self.object = self.parse_body()
                  ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/libcloud/common/google.py", line 267, in parse_body
    raise QuotaExceededError(message, self.status, code)
libcloud.common.google.QuotaExceededError: "Quota 'FIREWALLS' exceeded.  Limit: 500.0 globally."
2023-06-07T18:27:45Z <Greenlet at 0x7f056a3a2840: _up_async(<LayoutModel: 2>)> failed with QuotaExceededError

```

I tried running `mage integration:clean` and pulling the latest ogc image. Both had no effect.

**Comment by adam-stokes** [comment]
Those are the quota limits for the firewalls that are created within GCE, to unblock, someone with access to GCE admin can go in and delete those existing firewalls. I will look into making sure these are better handled so we do not hit that limitation.

**Comment by pierrehilbert** [comment]
Thanks @adam-stokes.
@alexsapran / @pazone could you help here in the meantime?

**Comment by alexsapran** [comment]
I provided access to the GCP project to @pazone in order to be able to clean up those rules.

@pierrehilbert can I ask to raise a ticket with eng-prod to increase those limits?

**Comment by pazone** [comment]
The firewall rules have been deleted. It's not a blocker now for a few days

**Comment by jlind23** [comment]
@adam-stokes were you able to look at this?

cc @cachedout

**Comment by adam-stokes** [comment]
I have a fix but I didn't want to push a new image until we could have this run. Would this project be able to run a different ogc image for a bit to see if that shows the same behavior after a month?

**Comment by blakerouse** [comment]
@adam-stokes Let me know the label to change it to and I will create a PR and link it to this issue with the tag name.

**Comment by jlind23** [comment]
@blakerouse please close this issue as soon as your PR with ogc:blake is merged. We will reopen it afterwards if needed.

**Comment by blakerouse** [comment]
PR is here: https://github.com/elastic/elastic-agent/pull/2884

**Comment by ycombinator** [comment]
> @blakerouse please close this issue as soon as your PR with ogc:blake is merged. We will reopen it afterwards if needed.

Actualy, could we leave this issue open even after #2884 is merged?  That way we remember to replace `ogc:blake` with `ogc:latest` in the code.

**Comment by blakerouse** [comment]
@ycombinator I will file another issue for that.
